### PR TITLE
Added , and / as separators for leader skill searching. 

### PR DESCRIPTION
**Pull request by muffin-rice**
what am i even doing

Quotes not needed for multi-word queries if the other is correctly parsed, ie ^ls "horus" revo horus works now. Probably fails if quotes or slashes are used as actual search terms.

### CODE REVIEW

File: padinfo/padinfo.py [hunk 1]
```diff
@@ -512,32 +513,80 @@ async def collabscroll(self, ctx, *, query: str):
             await ctx.send(self.makeFailureMsg(err))
 
 
+    async def _parse_ls_query(self, query : str) -> (str, str): 
+        '''Helper function for leaderskill. Takes in a query 
+        and returns the left or right'''
```
Comment (RheingoldRiver) [inline padinfo/padinfo.py:518]: can you make this one line and use `"""`

File: padinfo/padinfo.py [hunk 2]
```diff
@@ -512,32 +513,80 @@ async def collabscroll(self, ctx, *, query: str):
...
+        if len(words) >= 3: 
+            raise Exception('Too many inputs given. Limit to 1 or 2, or use quote wrappers or a separator (/ or ,) \
+to separate.') 
+        
+        #it's 2 words, no prefix
+        return (words[0], words[1])
+
     @commands.command(aliases=['leaders', 'leaderskills', 'ls'])
-    async def leaderskill(self, ctx, left_query: str, right_query: str = None, *, bad=None):
+    async def leaderskill(self, ctx, *, query : str):
```
Comment (RheingoldRiver) [inline padinfo/padinfo.py:567]: If you instead change this signature to:
```
async def leaderskill(self, ctx, left_query: str, right_query: str = None, *, rest_of_query=None):
```
and then immediately return when rest_of_query is None, you will have for free handled the case when quotes are used. Does this help?

File: padinfo/padinfo.py [hunk 3]
```diff
@@ -512,32 +513,80 @@ async def collabscroll(self, ctx, *, query: str):
...
+        '''Helper function for leaderskill. Takes in a query 
+        and returns the left or right'''
+        if not query: 
+            raise Exception('Query seems to be empty.')
+
+        # i assume neither '"' nor '/' will be used as part of card search
+        # this handles splitting with quotewrapping and /
+        # also comma splitting if there are quotes
+        if '"' in query or '/' in query or ',' in query: 
+            p = r'(?:(?:([^"/,]*))|(?:"([^"/]*)"))\s*(?:/|,)?\s*(?:(?:(?:([^"/,]*))|(?:"([^"/]*)")))?'
```
Comment (chasehult) [inline padinfo/padinfo.py:526]: `re.split(pattern, string)` might be a better way to do this.  This regex is kinda hard to parse manually!

File: padinfo/padinfo.py (hunk 2 above)
Comment (muffin-rice) [reply padinfo/padinfo.py:567]: Possibly, but that has the issue of the default space splitting; something like ^ls horus/revo horus would be harder with it.

Comment (RheingoldRiver) [reply padinfo/padinfo.py:567]: ah true, good point

Comment (RheingoldRiver) [reply padinfo/padinfo.py:567]: ok I think this is still the right signature. Then there's a few cases:

First, assume rest_of_query is None.

1. No delimiters in either `left_query` (lq) or `right_query` (rq). Great! Return.
2. A delimiter in lq but not rq, or the other way around. Then almost certainly the sender actually meant the delimiter to be the delimiter of the query. Let's check is it a comma and is there a space inside of this argument? If so then it's probably an exact name so let's respect it; if not then merge with case that rest_of_query is not None.
3. both have delimiters -> are they both exact names? if not then error.

If rest_of_query is not None then assume we had no quotes whatsoever and split on the delimiter using re.split.

**Comment by muffin-rice** [comment]
Reopening after my force push

File: padinfo/padinfo.py [hunk 4]
```diff
@@ -524,22 +524,33 @@ async def collabscroll(self, ctx, *, query: str):
...
         If either your left or right query contains spaces, wrap in quotes.
         e.g.: [p]leaderskill "r sonia" "b sonia"
         """
-        if bad:
-            await ctx.send(inline('Too many inputs. Try wrapping your queries in quotes.'))
-            return
+        # order of separators is '/' > '"' > ','
+        pref_query = []
+        if '/' in whole_query:
+            left_query, right_query = whole_query.split('/')[:2]
```
Comment (chasehult) [inline padinfo/padinfo.py:536]: one fancy thing you can do is
`left_query, right_query, *bad = whole_query.split('/')`
just to weed out bad input but this is totally optional. 

I would recommend doing that on the space one though.

**Review by chasehult — CHANGES_REQUESTED** [review]

**Review by chasehult — APPROVED** [review]
Approved
